### ISSUE TITLE
Direct messages sent while offline aren't recieved by slack-libpurple

### ISSUE THREAD

**Issue by aguenther** [dylex/slack-libpurple#85]
It seems like slack-libpurple either isn't recieving notifications about offline messages, or isn't processing them.

When I - once per week or so - remember to open slack in the browser, it shows me various unread notifications of people who tried to contact me. Pidgin with slack-libpurple installed doesn't show any kind notification, and seems to be unable to fetch the missed text.

I just built the current master and tested it again. Can reproduce.

**Comment by dylex** [comment]
Are these IMs or group messages? Are they sent while you're connected or not?

**Comment by aguenther** [comment]
IMs, sent while I'm not connected.

**Comment by aguenther** [comment]
I am trying to do some debugging on slack-libpurple myself right now. 
I tracked it to slack-im.c:94 by now, the conversation and user ID of the missing messages do end up in that debug output there.

What seems to be missing at this point is a call to conversations.history with the proper timestamps, to check whenever there are unread messages from that buddy, and if yes, add them as unread messages to pidgin. Or am I misunderstanding the whole thing here?

**Comment by dylex** [comment]
Yeah -- the problem with that approach is that it sends all the requests at once, which can make slack start denying requests, and eventually blacklist the client. I'm slowly working on a better solution on the `lazy` branch, but still trying to figure out the best way to deal with the unread counts.  What I really want is to have them show up on the buddy list, rather than open all the windows.  I think it should be pretty easy to just append an unread count label to the name in the buddy list (as long as the user hasn't set a custom alias), so I'm really tempted to do that. However, the community preference sounds like it is to open all the windows regardless, in which case these requests need to be thottled (serialized) somehow.
